src/wms/src/main/java/org/geoserver/wms/georss/AtomUtils.java

```diff
@@ -86,18 +86,22 @@ public static String dateToRFC3339(Date d){
 
     //TODO: use an html based output format
     public static String getEntryURL(WMS wms, SimpleFeature feature, WMSMapContent context){
-        String nsUri = feature.getType().getName().getNamespaceURI();
-        String nsPrefix = wms.getNameSpacePrefix(nsUri);
-
-        HashMap<String,String> params = new HashMap<String,String>();
-        params.put("format", "application/atom+xml");
-        params.put("layers",  nsPrefix + ":" + feature.getType().getTypeName());
-        params.put("featureid", feature.getID());
-        
-        return ResponseUtils.buildURL(context.getRequest().getBaseUrl(),
-             "wms/reflect",
-             params,
-             URLType.SERVICE);
+		try {
+			return featureTemplate.link(feature);
```

Comment:
Not respecting coding conventions. Please check and follow the guidelines for pull requests: https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md#commit-guidelines

Also check the rest of the document (for example, have you already sent a contribution agreement?, did the files need to have copyright headers updated?)